### PR TITLE
Implement ability to select specific wallet for packet clearing CLI

### PR DESCRIPTION
**Pull request by seanchen1991**
Closes: #2111 

## Description

Add `--key <KEY>` and `--counterparty-key <KEY>` options for clear packets CLI, just like Hermes provides one for `tx ft-transfer -key <KEY>`.

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Hermes) or unit/mock tests (for modules). 
- [x] Linked to GitHub issue.
- [x] Updated code comments and documentation (e.g., `docs/`). Add a definition of what keys are in the "keys" chapter of the Hermes guide. 

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

**Comment by seanchen1991** [comment]
It looks like as part of [ADR 10](https://github.com/informalsystems/ibc-rs/pull/2306) it was decided that CLI commands that require being able to specify parameters on both the source and destination chains should do so using `-host` and `-reference` suffixes. 

@adizere pointed out that the original design of this feature comes up a bit short because the `clear packet` flow sends messages to both a source and destination chain; specifying a single key via the flag doesn't make it clear which chain the key is associated with. 

The most straightforward way to address this is to split the proposed `key` flag into two flags that follow the convention laid out in the ADR. However, `--key-host` and `--key-reference` are confusing; `--host-key` and `--reference-key` are more appropriate. Though to be honest, "reference key" can also be confusing. 

Curious what folks thoughts are on how this question might be best resolved.

**Comment by adizere** [comment]
> The most straightforward way to address this is to split the proposed `key` flag into two flags that follow the convention laid out in the ADR. However, `--key-host` and `--key-reference` are confusing; `--host-key` and `--reference-key` are more appropriate. Though to be honest, "reference key" can also be confusing.

I agree this is not ideal. The "host" + "reference" terminology seems to be useful for client-related queries only.

Just to understand the current situation: We're refactoring ([cf ADR 10](https://github.com/informalsystems/ibc-rs/blob/master/docs/architecture/adr-010-unified-cli-arguments-hermes.md#clear-packets)) the `clear packets` CLI as follows:

> clear packets --chain <CHAIN_ID> --port <PORT_ID> --chan <CHANNEL_ID>

This refactor is underway here https://github.com/informalsystems/ibc-rs/pull/2275. Note that whatever changes we introduce in the present PR will likely conflict with that refactor, so we should aim to minimize merge conflicts, but that's a separate topic.

I looked again through the ADR and found that the specifiers we typically use as of ADR 10 are as follows:

> Example of specifiers we currently use are `host`, `reference`, `a`, `b` and `counterparty`.

If we want to have a specifier to a `--key` option, the specifier should probably be one of the above, to avoid introducing new ones. So we are left with two options from what I can tell: 
1. `--a-key` + `--b-key` or 
2. `--key` + `--counterparty-key`.

I doubt option 1 would be ideal here since `clear packets` CLI has no "a" or "b" options.

Additionally, we should also be consistent with the [key name specifiers from ADR 010](https://github.com/informalsystems/ibc-rs/blob/master/docs/architecture/adr-010-unified-cli-arguments-hermes.md#commands-for-keys). So it's more like

1. `--a-key-name` + `--b-key-name` or 
2. `--key-name` + `--counterparty-key-name`.

PS. should we update ADR 010 with whichever decision we take? to clearly specify how `clear packets` looks like.

**Comment by seanchen1991** [comment]
I agree that `a` and `b` specifiers in this case are not very helpful. It sounds like going with `--key-name` and `--counterparty-key-name` is the only viable option then. 

> should we update ADR 010 with whichever decision we take? to clearly specify how clear packets looks like.

Agree that the ADR should be updated.

File: relayer-cli/src/commands/clear.rs [hunk 1]
```diff
@@ -47,6 +78,14 @@ impl Runnable for ClearPacketsCmd {
             Err(e) => Output::error(format!("{}", e)).exit(),
         };
 
+        // Now that we have access to the counterparty chain at this point, we can
+        // match on `counterparty_key_name` and if a String exists, then override
+        // the counterparty chain's `key_name` in its own Config
+        if let Some(ref counterparty_key_name) = self.counterparty_key_name {
```
Comment (adizere) [inline relayer-cli/src/commands/clear.rs:84]: I think you need to call `config` on an object of type `ChainHandle`, cannot call the method on the trait directly.

One way to do it is using the `chains` variable defined above (L71).

```rust
let dst_chain_cfg = chains.dst.config();
```

**Review by romac — APPROVED** [review]

**Pull request merged**